Dark_Dusky_Positive_Selection

All of the Code in this repository was used for testing positive selection between S. variabilis and S. ciliatus. 

Trinity.sh is an example of the code used to assembly the raw reads for both species into transcriptomes.

Busco.sh is an example of the code that was used to check the transcriptomes for completedness.

CDhit.sh is and example of the code used to run CDhit.

Sort_Genes.py is the script that was used to separate the genes based off of EggNogMapper annotation. You must remove the header line before this script will run.

Translate.py is the script that was used to translate the ORFs.

Pal2nal.sh is the script that was used to create codon alignments.

FastTree.sh is the code that was used to create gene trees.

MakeForeground.py is the script that was used to select each species as the foreground for positive selection tests. 

Busted.sh is the script that was used to run Hyphy Busted.

Relax.sh is the script that was used to run Hyphy Relax.

Percent_Identity.py is the script that was used to calculate percent identity.

Graph_Ontology.R is the code that was used to make the bar plots for the publication.

